SME.CDEP.Aplicacao/Servicos/ServicoImportacaoArquivoBase.cs

```diff
@@ -590,5 +590,42 @@ private bool ObterValorBoleanoPorValorCampo(string valorDoCampo, string nomeDoCa
             }
             return false;
         }
+        
+        private async Task<ImportacaoArquivo> ObterImportacaoArquivoPorIdComValidacoes(long id,TipoAcervo tipoAcervoEsperado)
+        {
+            var arquivo = await repositorioImportacaoArquivo.ObterPorId(id);
+
+            if (arquivo.EhNulo())
+                throw new NegocioException(Constantes.ARQUIVO_NAO_ENCONTRADO);
+
+            if (arquivo.Conteudo.EhNulo() || arquivo.Conteudo.NaoEstaPreenchido())
+                throw new NegocioException(Constantes.CONTEUDO_DO_ARQUIVO_INVALIDO);
+
+            if (arquivo.TipoAcervo.NaoSaoIguais(tipoAcervoEsperado))
+                throw new NegocioException(string.Format(Constantes.ESSE_ARQUIVO_NAO_EH_ACERVO_X, tipoAcervoEsperado.Nome()));
+
+            return arquivo;
+        }
+        
+        public async Task<bool> RemoverLinhaDoArquivo<T>(long id, int linhaDoArquivo, TipoAcervo tipoAcervoEsperado) where T: AcervoLinhaDTO
+        {
+            var arquivo = await ObterImportacaoArquivoPorIdComValidacoes(id, tipoAcervoEsperado);
+            
+            var conteudo = JsonConvert.DeserializeObject<IEnumerable<T>>(arquivo.Conteudo);
+
+            var existeLinha = conteudo.Any(w => w.NumeroLinha.SaoIguais(linhaDoArquivo));
```

Comment:
Para evitar Code Smells de possível referencia nula :
```suggestion
            var existeLinha = conteudo?.Any(w => w.NumeroLinha.SaoIguais(linhaDoArquivo)) ?? false;
```